Dev branch

rsundar opened the pull request:
My partner finished the bottom half of the page, and i did the top.

rsundar committed:
prepared project structure, added top nav bar.

addod19 committed:
building layout

rsundar committed:
styling corrections.

rsundar committed:
added latest deals section and made hieghts and widths rems and percentages.

addod19 committed:
setup footer nd its styling

rsundar committed:
made changes to the nav bar, made it responsive.

rsundar committed:
Merge branch 'dev-branch' of https://www.github.com/rsundar/the-next-web into dev-branch

rsundar committed:
added article 1

rsundar committed:
styling changes.

addod19 committed:
created and input data in the sub-footer

rsundar committed:
made styling changes to top article header.

rsundar committed:
Merge branch 'dev-branch' of https://www.github.com/rsundar/the-next-web into dev-branch

rsundar committed:
made the top article section responsive.

rsundar committed:
added the top article section and associated styling.

addod19 committed:
done with the footer

addod19 committed:
styled the 'latest deals' and 'see all deals'

rsundar committed:
more styling changes to top article

rsundar committed:
resolving merge conflict

rsundar committed:
Revert "reverting to previous commit"

This reverts commit 3c8b0615c87d1434d89f8424eb6f32a29c33ad51, reversing
changes made to 94caafbf97ea9a69bece5391015d6139dd87bb81.

rsundar committed:
made all text responsive.

rsundar committed:
simple styling changes.

rsundar committed:
added latest news section

rsundar committed:
made header more responsive.

rsundar committed:
finished the latest news section and made it responsive.

rsundar committed:
minor styling issues

rsundar committed:
more minor styling issues

rsundar committed:
finished the latest funding section, and made it responsive.

rsundar committed:
minor spacing issues

rsundar committed:
final styling

rsundar committed:
final styling changes to page.

mosaaleb commented on Assets/stylesheets/styles.css:
```diff
@@ -0,0 +1,1255 @@
+
+.site-nav {
+    display: flex;
+    padding-left:0;
```
`site-nav` should have fixed position on medium and bigger screens.

rsundar replied:
in the media query for below 768px this is how the header which .site-nav is part of is styled:

header {
        margin-top:-0.8rem;
        height: 4rem;
        width: 99%;
        position: fixed;
        z-index: 2;
        will-change: width;
    }

mosaaleb commented on index.html:
```diff
@@ -0,0 +1,983 @@
+                <li class="hide-below-768">FUNDAMENTALS</li>
+                <li class="hide-below-768">WORK 2030</li>
+                <li class="right-nav"><i class="fas fa-search" style="font-size:1.5rem;"></i></li>
+                <li class="show-less-than-768"><i class="fas fa-bars" style="font-size: 1.5rem;color: #f42;"></i></li>
```
Try to avoid inline styling as possible.

rsundar replied:
will try and avoid it next time

mosaaleb reviewed (changes requested):
## Amazing job @rsundar, @addod19  :wave:

Your code is well structured, the UI is clean, the breakpoints are working as expected :clap:. Still, there are a few requirements needs be handled.
I have added some comments below to handle first before merging. 

#### Also: 
- [ ] Run the stickler checks to make sure that our code base has good style guidelines.
- [ ] Complete the `readme.md` file with its relevant information.

Please submit a new code review, after we handle these minor requirements.

Happy coding :computer:

addod19 committed:
Update README.md

readme updated

Ispirett commented:
Hey folks great job so far with this project :100: You are doing well. There are some issues that need addressing. check below. 

- [x] Around `855px`
![currentProject](https://user-images.githubusercontent.com/38801423/61504614-08e0de80-a9aa-11e9-87d9-9f5ab77f1474.png) please addresse this.

- [ ] Around `1450px` there is this issue 
![curent](https://user-images.githubusercontent.com/38801423/61504806-ab00c680-a9aa-11e9-8766-bbd2dba7d084.png)

- [x] your site should also be responsive up to `480px` its currently not. You are missing this in you head HTML tag ` <meta name="viewport" content="width=device-width, initial-scale=1.0">`

- [ ] ** Lunch and Distract section**  the article titles should be wrap in the `<article>` tag

- [ ] This section    **Latest Deals section**         should also have        `<article>` tags
 

These are some guidelines for the footer section

- [ ] It should contain two parts: top and bottom
- [ ] The top part should have social media icons vertically and horizontally aligned (The display property with a value of flex is recommended)
- [ ] The bottom part should have the TNW logo with a copyright paragraph
- [ ] The size of the icon should be 40px
- [ ] If possible please try and make the social icons more visible.

Hey, guys, I know this is a lot if you have any issues you can reach out to me on Slack. Please make changes and send for another review thanks :100:

rsundar commented:
Neither I nor my partner have monitors that are 1450px wide we cannot replicate that issue to fix it. 

**NOTE TO THE NEW REVIEWER PLEASE READ THIS COMMENT**

 WE ARE ALREADY 6 WEEKS IN, YOU CANNOT KEEP ASKING US TO MAKE INANE/IRRELEVANT CHANGES TO THE PROJECT WHICH NEITHER INCREASE OUR UNDERSTANDING OF THE SUBJECT MATTER NOR DOES IT ACTUALLY MAKE ANY DIFFERENCE IN THE PERFORMANCE OF THE PROJECT. 

THE FIRST REVIEWER ONLY HAD AN ISSUE WITH THE README FILE AND SOME MINOR HEADER RELATED ISSUES,

THE SECOND REVIEWER'S CHANGES LED TO A WHOLE DAYS WORK AND REQUIRED A LOT OF CHANGES TO THE CSS AND LED TO MORE THINGS BREAKING. 

PLEASE EXPLAIN YOUR REQUIRED CHANGES, IF WE ARE TO SPEND A WHOLE DAY LOOKING OVER AND MAKING CHANGES TO SUCH A MASSIVE CODE BASE, YOU NEED TO AT THE VERY LEAST EXPLAIN THE RATIONALE BEHIND THE CHANGES CLEARLY, A ONE LINE ORDER TO MAKE CHANGES AND CLICK CHECKBOXES IS NOT ACCEPTABLE, **PLEASE RESPECT OUR TIME AS WELL**. 

IF THEY WERE MICROVERSE GUIDELINES PLEASE PROVIDE A LINK TO THE SAME SO THAT WE CAN LOOK OVER THEM, SINCE THE ODIN PROJECT'S WEBSITE DOES NOT CONTAIN THE SAME.

KINDLY KEEP THIS IN MIND WHILE REVIEWING THE PROJECT AGAIN., THE POINT IS TO LEARN AND BROADEN OUR UNDERSTANDING OF THE SUBJECT MATTER DO NOT MIRE US IN UNEEDEED BUREAUCRATIC INANITIES WHICH ONLY SERVE TO CONFUSE AND NOT CLARIFY. 

THANK YOU

rsundar committed:
repaired the appearence at 855px.

Ispirett commented:
> Neither I nor my partner have monitors that are 1450px wide we cannot replicate that issue to fix it.
> 
> **NOTE TO THE NEW REVIEWER PLEASE READ THIS COMMENT**
> 
> WE ARE ALREADY 6 WEEKS IN, YOU CANNOT KEEP ASKING US TO MAKE INANE/IRRELEVANT CHANGES TO THE PROJECT WHICH NEITHER INCREASE OUR UNDERSTANDING OF THE SUBJECT MATTER NOR DOES IT ACTUALLY MAKE ANY DIFFERENCE IN THE PERFORMANCE OF THE PROJECT.
> 
> THE FIRST REVIEWER ONLY HAD AN ISSUE WITH THE README FILE AND SOME MINOR HEADER RELATED ISSUES,
> 
> THE SECOND REVIEWER'S CHANGES LED TO A WHOLE DAYS WORK AND REQUIRED A LOT OF CHANGES TO THE CSS AND LED TO MORE THINGS BREAKING.
> 
> PLEASE EXPLAIN YOUR REQUIRED CHANGES, IF WE ARE TO SPEND A WHOLE DAY LOOKING OVER AND MAKING CHANGES TO SUCH A MASSIVE CODE BASE, YOU NEED TO AT THE VERY LEAST EXPLAIN THE RATIONALE BEHIND THE CHANGES CLEARLY, A ONE LINE ORDER TO MAKE CHANGES AND CLICK CHECKBOXES IS NOT ACCEPTABLE, **PLEASE RESPECT OUR TIME AS WELL**.
> 
> IF THEY WERE MICROVERSE GUIDELINES PLEASE PROVIDE A LINK TO THE SAME SO THAT WE CAN LOOK OVER THEM, SINCE THE ODIN PROJECT'S WEBSITE DOES NOT CONTAIN THE SAME.
> 
> KINDLY KEEP THIS IN MIND WHILE REVIEWING THE PROJECT AGAIN., THE POINT IS TO LEARN AND BROADEN OUR UNDERSTANDING OF THE SUBJECT MATTER DO NOT MIRE US IN UNEEDEED BUREAUCRATIC INANITIES WHICH ONLY SERVE TO CONFUSE AND NOT CLARIFY.
> 
> THANK YOU

I understand you may be feeling frustrated I do at times. But as a developler, you are always going  to be making changes, especially in the job environment. Microverse is just preparing you for the real job world. To address some of your concerns.

**1** you can use your dev tools in the browser to change screen sizes, It will mimic how your website will look on that selected resolution. 
![resize](https://user-images.githubusercontent.com/38801423/61560503-776d7d00-aa3a-11e9-8032-72436c7483c6.png). On the bottom left where the icon is highlighted in blue gives you the option to do so.

**2**  These guides are actual Mircoverse's guidelines for the project and they are working hard to get them available for students to make it easier. 

**3** The guidelines are not there to frustrate but to guide you. You are still in control of your project. In case there were things that you didn't understand I offered my assistance via slack out of my personal time.  

**4** While resubmitting a project plenty times may seem daunting, Just remember we are here to help and it's not always going to be easy but at least you have good support at Microverse.

rsundar commented:
> > Neither I nor my partner have monitors that are 1450px wide we cannot replicate that issue to fix it.
> > **NOTE TO THE NEW REVIEWER PLEASE READ THIS COMMENT**
> > WE ARE ALREADY 6 WEEKS IN, YOU CANNOT KEEP ASKING US TO MAKE INANE/IRRELEVANT CHANGES TO THE PROJECT WHICH NEITHER INCREASE OUR UNDERSTANDING OF THE SUBJECT MATTER NOR DOES IT ACTUALLY MAKE ANY DIFFERENCE IN THE PERFORMANCE OF THE PROJECT.
> > THE FIRST REVIEWER ONLY HAD AN ISSUE WITH THE README FILE AND SOME MINOR HEADER RELATED ISSUES,
> > THE SECOND REVIEWER'S CHANGES LED TO A WHOLE DAYS WORK AND REQUIRED A LOT OF CHANGES TO THE CSS AND LED TO MORE THINGS BREAKING.
> > PLEASE EXPLAIN YOUR REQUIRED CHANGES, IF WE ARE TO SPEND A WHOLE DAY LOOKING OVER AND MAKING CHANGES TO SUCH A MASSIVE CODE BASE, YOU NEED TO AT THE VERY LEAST EXPLAIN THE RATIONALE BEHIND THE CHANGES CLEARLY, A ONE LINE ORDER TO MAKE CHANGES AND CLICK CHECKBOXES IS NOT ACCEPTABLE, **PLEASE RESPECT OUR TIME AS WELL**.
> > IF THEY WERE MICROVERSE GUIDELINES PLEASE PROVIDE A LINK TO THE SAME SO THAT WE CAN LOOK OVER THEM, SINCE THE ODIN PROJECT'S WEBSITE DOES NOT CONTAIN THE SAME.
> > KINDLY KEEP THIS IN MIND WHILE REVIEWING THE PROJECT AGAIN., THE POINT IS TO LEARN AND BROADEN OUR UNDERSTANDING OF THE SUBJECT MATTER DO NOT MIRE US IN UNEEDEED BUREAUCRATIC INANITIES WHICH ONLY SERVE TO CONFUSE AND NOT CLARIFY.
> > THANK YOU
> 
> I understand you may be feeling frustrated I do at times. But as a developler, you are always going to be making changes, especially in the job environment. Microverse is just preparing you for the real job world. To address some of your concerns.
> 
> **1** you can use your dev tools in the browser to change screen sizes, It will mimic how your website will look on that selected resolution.
> ![resize](https://user-images.githubusercontent.com/38801423/61560503-776d7d00-aa3a-11e9-8032-72436c7483c6.png). On the bottom left where the icon is highlighted in blue gives you the option to do so.
> 
> **2** These guides are actual Mircoverse's guidelines for the project and they are working hard to get them available for students to make it easier.
> 
> **3** The guidelines are not there to frustrate but to guide you. You are still in control of your project. In case there were things that you didn't understand I offered my assistance via slack out of my personal time.
> 
> **4** While resubmitting a project plenty times may seem daunting, Just remember we are here to help and it's not always going to be easy but at least you have good support at Microverse.

Please provide us a rationale or link to those guidelines. We are still stuck on the html and css sections because the reviewers keep asking us to make wholesale changes which neither effect the performance of the page nor does it alter the appearance of the page.

Secondly, I have worked as an embedded developer, so I am aware of the SDLC no senior developer asks you to make changes without explaining why those changes are being made and how they would affect our release time or why certain code constructs are bad for the architecture we have chosen for a particular project. This is the mentorship role a senor dev takes up to show the ropes to his juniors. 

What you have asked for is not explained, they do not highlight fallacies in our understanding, they don not affect the appearance of the page beyond the errors you have highlighted. This is not an acceptable review of the codebase. 

**1) PLEASE EXPLAIN THE RATIONALE BEHIND SPLITTING THE FOOTER UP**

**2) THESE GUIDELINES DO FRUSTRATE US AS WE DO NOT UNDERSTAND WHY THEY ARE BEING ASKED**

**3) PROVIDE THE LINK TO REPO/PAGE WHERE THE MICROVERSE GUIDELINES FOR THE PROJECT ARE CLEARLY AND SUCCINCTLY WRITTEN UP**

**4) WHY ARE YOU ASKING US TO USE MAGIC NUMBERS (logo being 40px) IN A RESPONSIVE WEBSITE?**

**5) WHAT DOES MORE VISIBILITY FOR THE SOCIAL ICONS MEAN IT IS ENTIRELY SUBJECTIVE**

**6) WHY DO WE NEED ARTICLE TAGS FOR SECTIONS WHICH ARE NOT ARTICLES?**

Thank you

jofer86 commented:
**Follow up underway**

jofer86 commented:
## Hello team, as I can see you are in at discomfort I'll try to address your concerns one by one.

**1) PLEASE EXPLAIN THE RATIONALE BEHIND SPLITTING THE FOOTER UP** 
 Since we are making a clone website, functionality, as well as mark up, should follow the same as the site we are cloning. If you inspect the [site](https://thenextweb.com/) we are cloning you will see that, in fact, the footer section is divided in two parts `<div class="footer-links">` and `<div class="footer-legal">`

**2) THESE GUIDELINES DO FRUSTRATE US AS WE DO NOT UNDERSTAND WHY THEY ARE BEING ASKED**
Guidelines are aked by the Microverse staff because they expect that we code following a certain standard, they are not there to make our path more complicated, but to try to make our code better, more readable and understandable.

**3)PROVIDE THE LINK TO REPO/PAGE WHERE THE MICROVERSE GUIDELINES FOR THE PROJECT ARE CLEARLY AND SUCCINCTLY WRITTEN UP**
The Microverse staff, give us the TSEs the guidelines which we should follow along and see that they are met and ask for changes when deemed necessary, Is up to the Microverse staff to release those guidelines to students.

**4) WHY ARE YOU ASKING US TO USE MAGIC NUMBERS (logo being 40px) IN A RESPONSIVE WEBSITE?** 
As we are working on a clone site, it is expected to resemble as much as possible to the original site, in this case, if you inspect the icons on the footer section, the one that is enclosed in an anchor tag, you can see that the size is, in fact, 40px by 40px.

**5) WHAT DOES MORE VISIBILITY FOR THE SOCIAL ICONS MEAN IT IS ENTIRELY SUBJECTIVE**
As it is, in fact, something subjective, the icons used on the project do not have enough contrast with the background not making them stand out as in the original site. Like I said earlier we are building a clone of a site and we should try to mimic the functionality as much as we can.

**6) WHY DO WE NEED ARTICLE TAGS FOR SECTIONS WHICH ARE NOT ARTICLES?** 
This is a part of a semantic HTML markup, that might cause some confusion, you can check [this site](http://html5doctor.com/the-article-element/) for some clarification.

Hope this gives some clarification on the matter, from what I can see, and the observations made by my fellow TSE, I can see that the footer section is almost done, with the small exception of making the social icons the correct size as required by the Microverse Guidelines, and maybe changing the color to white to make them stand out.

In the section Latests news It seems only one image is showing, the one that shows the virtual reality headset.

The `<article class="img-container-">`tags do not have the property of no-repeat, so when the size of the screen gets bigger we start getting repeated images, please kindly fix that.

The same goes for the Launch and Distract background images.

When the screen width goes beyond 1200px the only section that should continue to keep on growing is the first container. The rest of the site should have a max-width of approximately 1024px.

Also we need to make sure that the stickler checks are in the green, so please make the neccesary adjustments as well as running your markup through a validator such as [this one](https://validator.w3.org/), and make sure is also in the green.


Kindly make the suggested changes and ask for a new code review.


## Cheers, Jorge F.

nidalaa commented:
Hello @rsundar @jofer86 @Ispirett and @mosaaleb 👋 

As this conversation was moved to many other communication channels and discussed with all of you, let me just leave a quick summary here:

- The fixed size for the icons in the footer indeed was not necessary to clone the website (we will make sure to never ask for it again).
- I can see commit which addresses some of the issues raised by @Ispirett  and @jofer86 👍 . @rsundar please take the rest of them as feedback, if you want to make your website look professional in 💯 %. And please keep in mind that in a real-life project you will be asked multiple times to make your CSS and HTML perfectly matching the design mockup.
![css](https://user-images.githubusercontent.com/7466903/61971502-155dcc00-af94-11e9-9826-eb5d64cc8a70.gif)

- @rsundar in order to merge your Pull Request to master, you will also need to resolve merge conflicts and lint errors. As you can see above it is not possible now ☝️ .
- @jofer86 feel free to mark this project as completed once it is ready to merge.

**Nice job everybody, I can see that everyone here wants to make all the projects as beautiful as possible.** 👏

rsundar commented:
> Hello @rsundar @jofer86 @Ispirett and @mosaaleb 
> 
> As this conversation was moved to many other communication channels and discussed with all of you, let me just leave a quick summary here:
> 
> * The fixed size for the icons in the footer indeed was not necessary to clone the website (we will make sure to never ask for it again).
> * I can see commit which addresses some of the issues raised by @Ispirett  and @jofer86 +1 . @rsundar please take the rest of them as feedback, if you want to make your website look professional in 100 %. And please keep in mind that in a real-life project you will be asked multiple times to make your CSS and HTML perfectly matching the design mockup.
>   ![css](https://user-images.githubusercontent.com/7466903/61971502-155dcc00-af94-11e9-9826-eb5d64cc8a70.gif)
> * @rsundar in order to merge your Pull Request to master, you will also need to resolve merge conflicts and lint errors. As you can see above it is not possible now point_up .
> * @jofer86 feel free to mark this project as completed once it is ready to merge.
> 
> **Nice job everybody, I can see that everyone here wants to make all the projects as beautiful as possible.**

Thank you so much @nidalaa 

We will work on the merge conflicts, stickler checks and some of the other issues that were mentioned by @jofer86 and submit for review again.

addod19 commented:
kindly send me the link to your repo


On Tue, Jul 30, 2019 at 10:20 AM Rohan Sundar <notifications@github.com>
wrote:

> Merged #1 <https://github.com/rsundar/the-next-web/pull/1> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rsundar/the-next-web/pull/1?email_source=notifications&email_token=ALB5HEL5TOJRV5XKAE7NEOLQCAIWRA5CNFSM4IEBSAZ2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOSYUVHGA#event-2519290776>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALB5HEJDJW4OQ3ITYG4L5HTQCAIWRANCNFSM4IEBSAZQ>
> .
>
